src/modules/campaign/ContentfulClient.spec.ts

```diff
@@ -1,87 +1,194 @@
+      })
+    })
+
+    describe('and the request fails', () => {
+      beforeEach(() => {
+        nock(CMS_URL)
+          .get('/spaces/space-id/environments/env-id/entries/entry-id/')
+          .query({ locale: 'en-US' })
+          .reply(500)
+      })
+
+      it('should throw an error', async () => {
+        await expect(
+          client.fetchEntry('space-id', 'env-id', 'entry-id')
+        ).rejects.toThrow('Failed to fetch entity data')
+      })
+    })
+  })
+
+  describe('fetchEntryAllLocales', () => {
+    it('should combine fields from all available locales', async () => {
+      const mockResponses = {
+        enUS: {
+          fields: { title: { [ContentfulLocale.enUS]: 'English Title' } }
+        },
+        es: {
+          fields: { title: { [ContentfulLocale.es]: 'Título en Español' } }
+        },
+        zh: { fields: { title: { [ContentfulLocale.zh]: '中文标题' } } }
+      }
+
+      global.fetch = jest
+        .fn()
+        .mockResolvedValueOnce({
           ok: true,
-          json: () => Promise.resolve(mockResponse)
+          json: () => Promise.resolve(mockResponses.enUS)
         })
+        .mockResolvedValueOnce({
+          ok: true,
+          json: () => Promise.resolve(mockResponses.es)
+        })
+        .mockResolvedValueOnce({
+          ok: true,
+          json: () => Promise.resolve(mockResponses.zh)
+        })
+
+      const result = await client.fetchEntryAllLocales(
+        mockSpace,
+        mockEnvironment,
+        mockId
+      )
+
+      expect(result).toEqual({
+        title: {
+          [ContentfulLocale.enUS]: 'English Title',
+          [ContentfulLocale.es]: 'Título en Español',
+          [ContentfulLocale.zh]: '中文标题'
+        }
       })
+    })
+  })
+
+  describe('fetchAsset', () => {
+    it('should fetch and transform an asset to the correct format', async () => {
+      global.fetch = jest.fn().mockResolvedValue({
+        ok: true,
+        json: () => Promise.resolve(marketplaceHomepageBannerAssets[0])
+      })
+
+      const result = await client.fetchAsset(mockSpace, mockEnvironment, mockId)
+
+      expect(result.fields).toHaveProperty('title.en-US')
+      expect(result.fields).toHaveProperty('description.en-US')
+      expect(result.fields).toHaveProperty('file.en-US')
+    })
+  })
 
-      it('should have fetched the entry with the given parameters', async () => {
-        await client.fetchEntry(
-          mockSpace,
-          mockEnvironment,
-          mockId,
-          mockContentType,
-          mockToken
-        )
-
-        expect(fetch).toHaveBeenCalledWith(
-          `https://cdn.contentful.com/spaces/${mockSpace}/environments/${mockEnvironment}/entries/?sys.id=${mockId}&content_type=${mockContentType}&locale=*`,
-          {
-            headers: {
-              Authorization: `Bearer ${mockToken}`
+  describe('fetchAssetsFromEntryFields', () => {
+    it('should fetch all assets referenced in the entry fields', async () => {
```

Comment:
Would you mind using the describe to explain each context?
```suggestion
   describe("when there are fields to get the assets from", () =>
    it('should get all assets referenced in the entry fields', async () => {
```